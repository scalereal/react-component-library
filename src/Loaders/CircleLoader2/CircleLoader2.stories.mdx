import CircleLoader2 from '.';
import { Preview, Story, Meta, Props, ArgsTable, Canvas } from '@storybook/addon-docs/blocks';
import { withKnobs, color, boolean, select } from '@storybook/addon-knobs';
import { SizeMap } from '../types';

<Meta
    title="Components/Loaders/CircleLoader2"
    component={CircleLoader2}
    argTypes={{
        size: {
            control: {
                type: 'select',
                options: Object.values(SizeMap)
            }
        },
        Color: {
            control: 'color'
        },
        display: {
            control: {
                type: 'boolean'
            }
        },
        ref: {
            table: {
                disable: true
            }
        },
        theme: {
            table: {
                disable: true
            }
        },
        as: {
            table: {
                disable: true
            }
        }
    }}
/>

# Loaders

Displays a Loader.

export const Template = args => (
    <div>
        <CircleLoader2 {...args} />
    </div>
);

<Preview>
    <Story name="CircleLoader2" args={{ size: 'S' }}>
        {Template.bind({})}
    </Story>
</Preview>

### Color Variations

You can either change the theme or pass the colors as props.

<Preview withToolbar>
    <div style={{ display: 'flex', justifyContent: 'space-around' }}>
        <CircleLoader2 size="S" Color="green" />
        <CircleLoader2 size="S" Color="red" />
    </div>
</Preview>

### Sizes

There are Six sizs available `XXS` | `XS` | `S` | `M` | `L` | `XL`.

<Preview withToolbar>
    <div style={{ display: 'flex', justifyContent: 'space-between' }}>
        <DotsCircle size="XXS" />
        <CircleLoader2 size="XS" />
        <CircleLoader2 size="S" />
        <CircleLoader2 size="M" />
        <CircleLoader2 size="L" />
        <CircleLoader2 size="XL" />
    </div>
</Preview>

### Props

<Props story="CircleLoader2" />
