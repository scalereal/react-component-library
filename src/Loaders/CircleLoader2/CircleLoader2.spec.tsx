import { render } from '@testing-library/react';
import * as React from 'react';
import CircleLoader2 from '.';

describe('CircleLoader2 Loader', () => {
    it('should render with default theme', () => {
        const { container } = render(<CircleLoader2/>);
        expect(container).toMatchSnapshot();
    });
    it('should render with all the props given', () => {
        const { container } = render(
            <CircleLoader2
                size="M"
                Color='#126AFA'
            />
        );
        expect(container).toMatchSnapshot();
    });

    it('should render image avatar properly', () => {
        const { container } = render(
            <CircleLoader2 size="L" />
        );
        expect(container).toMatchSnapshot();
    });

});