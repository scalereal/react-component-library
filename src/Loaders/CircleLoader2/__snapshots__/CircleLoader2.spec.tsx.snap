// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CircleLoader2 Loader should render image avatar properly 1`] = `
<div>
  <div
    class="sc-gsTCUz gvGlZC"
  >
    <div
      class="sc-bdfBwQ hefCnw"
    />
  </div>
</div>
`;

exports[`CircleLoader2 Loader should render with all the props given 1`] = `
<div>
  <div
    class="sc-gsTCUz eYYxuo"
  >
    <div
      class="sc-bdfBwQ kCXDKV"
    />
  </div>
</div>
`;

exports[`CircleLoader2 Loader should render with default theme 1`] = `
<div>
  <div
    class="sc-gsTCUz htzeJZ"
  >
    <div
      class="sc-bdfBwQ hefCnw"
    />
  </div>
</div>
`;
